rust_snuba/src/processors/mod.rs

```diff
@@ -61,6 +69,16 @@ pub fn get_cogs_label(processor_name: &str) -> Option<String> {
     }
 }
 
+pub fn get_processing_function_with_replacements(
```

Comment:
I think having a new kind of processor function here is tricky because it means that every test we have that tests all message processors generically has to be updated (such as `test_schemas` in this file)

is there no way to make all message processors return the same type?

Reply:
I tweaked things slightly. Processors still return two different types unfortunately, but I'm defining them in one spot now, and I updated `test_schemas` to test the two kinds of processors (with and without replacements). Let me know if you think this is an ok compromise.